lib/evmone/instructions.hpp

```diff
@@ -269,20 +269,15 @@ inline evmc_status_code shr(ExecutionState& state) noexcept
 
 inline evmc_status_code sar(ExecutionState& state) noexcept
 {
-    if ((state.stack[1] & (uint256{1} << 255)) == 0)
-        return shr(state);
+    const auto y = state.stack.pop();
+    auto& x = state.stack.top();
 
-    constexpr auto allones = ~uint256{};
+    const bool is_neg = static_cast<int64_t>(x[3]) < 0;
```

Comment:
Why is this looking at the lowest 64-bit (BE)?

Reply:
This is the highest. Words are in LE.

Reply:
Can you leave a comment in the code?

Reply:
Added comment.